code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -153,12 +169,72 @@
 	empty_contents()
 	if(!objective_complete)
 		destroy_rifts()
+	add_overlay()
 	..()
 
 /mob/living/simple_animal/hostile/space_dragon/wabbajack_act(mob/living/new_mob)
 	empty_contents()
 	. = ..()
 
+/**
+  * Allows space dragon to choose its own name.
+  *
+  * Prompts the space dragon to choose a name, which it will then apply to itself.
+  * If the name is invalid, will re-prompt the dragon until a proper name is chosen.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/dragon_name()
+	var/chosen_name = sanitize_name(reject_bad_text(stripped_input(src, "What would you like your name to be?", "Choose Your Name", real_name, MAX_NAME_LEN)))
+	if(!chosen_name)
+		to_chat(src, "<span class='warning'>Not a valid name, please try again.</span>")
+		dragon_name()
+		return
+	visible_message("<span class='notice'>Your name is now <span class='name'>[chosen_name]</span>, the feared Space Dragon.</span>")
+	fully_replace_character_name(null, chosen_name)
+
+/**
+  * Allows space dragon to choose a color for itself.
+  *
+  * Prompts the space dragon to choose a color, from which it will then apply to itself.
+  * If an invalid color is given, will re-prompt the dragon until a proper color is chosen.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/color_selection()
+	chosen_color = input(src,"What would you like your color to be?","Choose Your Color","#ffffff") as color|null
+	if(!chosen_color) //redo proc until we get a color
+		to_chat(src, "<span class='warning'>Not a valid color, please try again.</span>")
+		color_selection()
+		return
+	var/temp_hsv = RGBtoHSV(chosen_color)
+	if(chosen_color == "#000000")
+		chosen_color = "#ffffff"
+	else if(ReadHSV(temp_hsv)[3] < ReadHSV("#7F7F7F")[3])
```

Comment:
Use the color defines, here and everywhere else.

Also, I'd rather instead of using `ReadHSV("#7F7F7F")[3]`, you just have a file specific define for the exact value threshold.